### ISSUE TITLE
Evaluation results are always zero,fvcore version

### ISSUE THREAD

**Issue by ZitengXue** [UARK-AICV/AISFormer#6]
Hi @trqminh !
    I'm appreciated for your impressive work!
    After configuring everything(KINS) as your instructions(During the generating the instance_train_amodal.json file, I didn't use the [annotation json file update 2020](https://drive.google.com/drive/folders/1FuXz1Rrv5rrGG4n7KcQHVWKvSyr3Tkyo?usp=sharing) cause I found that it have 8 classes:meta.thing_classes =['cyclist', 'pedestrian', 'rider', 'car', 'tram', 'truck', 'van', 'misc']),but  it's 7 classes  in the dataset [builtin.py](https://github.com/UARK-AICV/AISFormer/blob/master/detectron2/data/datasets/builtin.py)  meta.thing_classes =['cyclist', 'pedestrian', 'car', 'tram', 'truck', 'van', 'misc'],so I use the original KINS datasets annoation json file to generate the instance_train_amodal.json
    Then I run [scripts/train.sh](https://github.com/UARK-AICV/AISFormer/blob/master/scripts),though everything seems goes well,but  I the AP is always 0

[04/06 16:04:54 fvcore.common.checkpoint]: [Checkpointer] Loading from detectron2://ImageNetPretrained/MSRA/R-50.pkl ...

[04/05 21:35:04] fvcore.common.checkpoint WARNING: Some model parameters or buffers are not found in the checkpoint:
backbone.fpn_lateral2.{bias, weight}
backbone.fpn_lateral3.{bias, weight}
backbone.fpn_lateral4.{bias, weight}
backbone.fpn_lateral5.{bias, weight}
backbone.fpn_output2.{bias, weight}
backbone.fpn_output3.{bias, weight}
backbone.fpn_output4.{bias, weight}
backbone.fpn_output5.{bias, weight}
proposal_generator.rpn_head.anchor_deltas.{bias, weight}
proposal_generator.rpn_head.conv.{bias, weight}
proposal_generator.rpn_head.objectness_logits.{bias, weight}
roi_heads.box_head.fc1.{bias, weight}
roi_heads.box_head.fc2.{bias, weight}
roi_heads.box_predictor.bbox_pred.{bias, weight}
roi_heads.box_predictor.cls_score.{bias, weight}
roi_heads.mask_head.mask_head_model.deconv_for_TR.0.{bias, weight}
roi_heads.mask_head.mask_head_model.mask_embed.layers.0.{bias, weight}
roi_heads.mask_head.mask_head_model.mask_embed.layers.1.{bias, weight}
roi_heads.mask_head.mask_head_model.mask_embed.layers.2.{bias, weight}
roi_heads.mask_head.mask_head_model.mask_feat_learner_TR.{bias, weight}
roi_heads.mask_head.mask_head_model.norm_rois.{bias, weight}
roi_heads.mask_head.mask_head_model.pixel_embed.{bias, weight}
roi_heads.mask_head.mask_head_model.positional_encoding.col_embed.weight
roi_heads.mask_head.mask_head_model.positional_encoding.row_embed.weight
roi_heads.mask_head.mask_head_model.predictor_TR.{bias, weight}
roi_heads.mask_head.mask_head_model.query_embed.weight
roi_heads.mask_head.mask_head_model.subtract_model.layers.0.{bias, weight}
roi_heads.mask_head.mask_head_model.subtract_model.layers.1.{bias, weight}
roi_heads.mask_head.mask_head_model.transformer_decoder.layers.0.linear1.{bias, weight}
roi_heads.mask_head.mask_head_model.transformer_decoder.layers.0.linear2.{bias, weight}
roi_heads.mask_head.mask_head_model.transformer_decoder.layers.0.multihead_attn.out_proj.{bias, weight}
roi_heads.mask_head.mask_head_model.transformer_decoder.layers.0.multihead_attn.{in_proj_bias, in_proj_weight}
roi_heads.mask_head.mask_head_model.transformer_decoder.layers.0.norm1.{bias, weight}
roi_heads.mask_head.mask_head_model.transformer_decoder.layers.0.norm2.{bias, weight}
roi_heads.mask_head.mask_head_model.transformer_decoder.layers.0.norm3.{bias, weight}
roi_heads.mask_head.mask_head_model.transformer_decoder.layers.0.self_attn.out_proj.{bias, weight}
roi_heads.mask_head.mask_head_model.transformer_decoder.layers.0.self_attn.{in_proj_bias, in_proj_weight}
roi_heads.mask_head.mask_head_model.transformer_encoder.layers.0.linear1.{bias, weight}
roi_heads.mask_head.mask_head_model.transformer_encoder.layers.0.linear2.{bias, weight}
roi_heads.mask_head.mask_head_model.transformer_encoder.layers.0.norm1.{bias, weight}
roi_heads.mask_head.mask_head_model.transformer_encoder.layers.0.norm2.{bias, weight}
roi_heads.mask_head.mask_head_model.transformer_encoder.layers.0.self_attn.out_proj.{bias, weight}
roi_heads.mask_head.mask_head_model.transformer_encoder.layers.0.self_attn.{in_proj_bias, in_proj_weight}

[04/05 21:35:04] fvcore.common.checkpoint WARNING: The checkpoint state_dict contains keys that are not used by the model:
  fc1000.{bias, weight}
  stem.conv1.bias
[04/05 21:35:04] d2.engine.train_loop INFO: Starting training from iteration 0
[04/05 21:35:08] d2.data.datasets.coco INFO: Loading /home/xzt/ASIF/data/KINS/instances_val.json takes 1.90 seconds.
[04/05 21:35:08] d2.data.datasets.coco WARNING: 
Category ids in annotations are not in [1, #categories]! We'll apply a mapping for you.

[04/05 22:34:34] d2.engine.defaults INFO: Evaluation results for kins_dataset_val in csv format:
[04/05 22:34:34] d2.evaluation.testing INFO: copypaste: Task: amodal_segm
[04/05 22:34:34] d2.evaluation.testing INFO: copypaste: AP,AP50,AP75,APs,APm,APl,AR1,AR10,AR100
[04/05 22:34:34] d2.evaluation.testing INFO: copypaste: 0.0000,0.0000,0.0000,0.0001,0.0003,0.0006,0.0000,0.0074,0.0442
[04/05 22:34:34] d2.evaluation.testing INFO: copypaste: Task: bbox
[04/05 22:34:34] d2.evaluation.testing INFO: copypaste: AP,AP50,AP75,APs,APm,APl
[04/05 22:34:34] d2.evaluation.testing INFO: copypaste: 0.0000,0.0000,0.0000,0.0001,0.0003,0.0007
[04/05 22:34:34] d2.evaluation.testing INFO: copypaste: Task: visible_segm
[04/05 22:34:34] d2.evaluation.testing INFO: copypaste: AP,AP50,AP75,APs,APm,APl,AR1,AR10,AR100

**Comment by Chjdl1816** [comment]
There is the same warning when training the cocoa data set, but from the convergence of the loss function, it seems that the training is no problem

WARNING [04/06 19:40:57 fvcore.common.checkpoint]: Some model parameters or buffers are not found in the checkpoint:
backbone.fpn_lateral2.{bias, weight}
backbone.fpn_lateral3.{bias, weight}
backbone.fpn_lateral4.{bias, weight}
backbone.fpn_lateral5.{bias, weight}
backbone.fpn_output2.{bias, weight}
backbone.fpn_output3.{bias, weight}
backbone.fpn_output4.{bias, weight}
backbone.fpn_output5.{bias, weight}
proposal_generator.rpn_head.anchor_deltas.{bias, weight}
proposal_generator.rpn_head.conv.{bias, weight}
proposal_generator.rpn_head.objectness_logits.{bias, weight}
roi_heads.box_head.fc1.{bias, weight}
roi_heads.box_head.fc2.{bias, weight}
roi_heads.box_predictor.bbox_pred.{bias, weight}
roi_heads.box_predictor.cls_score.{bias, weight}
roi_heads.mask_head.mask_head_model.deconv_for_TR.0.{bias, weight}
roi_heads.mask_head.mask_head_model.mask_embed.layers.0.{bias, weight}
roi_heads.mask_head.mask_head_model.mask_embed.layers.1.{bias, weight}
roi_heads.mask_head.mask_head_model.mask_embed.layers.2.{bias, weight}
roi_heads.mask_head.mask_head_model.mask_feat_learner_TR.{bias, weight}
roi_heads.mask_head.mask_head_model.norm_rois.{bias, weight}
roi_heads.mask_head.mask_head_model.pixel_embed.{bias, weight}
roi_heads.mask_head.mask_head_model.positional_encoding.col_embed.weight
roi_heads.mask_head.mask_head_model.positional_encoding.row_embed.weight
roi_heads.mask_head.mask_head_model.predictor_TR.{bias, weight}
roi_heads.mask_head.mask_head_model.query_embed.weight
roi_heads.mask_head.mask_head_model.subtract_model.layers.0.{bias, weight}
roi_heads.mask_head.mask_head_model.subtract_model.layers.1.{bias, weight}
roi_heads.mask_head.mask_head_model.transformer_decoder.layers.0.linear1.{bias, weight}
roi_heads.mask_head.mask_head_model.transformer_decoder.layers.0.linear2.{bias, weight}
roi_heads.mask_head.mask_head_model.transformer_decoder.layers.0.multihead_attn.out_proj.{bias, weight}
roi_heads.mask_head.mask_head_model.transformer_decoder.layers.0.multihead_attn.{in_proj_bias, in_proj_weight}
roi_heads.mask_head.mask_head_model.transformer_decoder.layers.0.norm1.{bias, weight}
roi_heads.mask_head.mask_head_model.transformer_decoder.layers.0.norm2.{bias, weight}
roi_heads.mask_head.mask_head_model.transformer_decoder.layers.0.norm3.{bias, weight}
roi_heads.mask_head.mask_head_model.transformer_decoder.layers.0.self_attn.out_proj.{bias, weight}
roi_heads.mask_head.mask_head_model.transformer_decoder.layers.0.self_attn.{in_proj_bias, in_proj_weight}
roi_heads.mask_head.mask_head_model.transformer_encoder.layers.0.linear1.{bias, weight}
roi_heads.mask_head.mask_head_model.transformer_encoder.layers.0.linear2.{bias, weight}
roi_heads.mask_head.mask_head_model.transformer_encoder.layers.0.norm1.{bias, weight}
roi_heads.mask_head.mask_head_model.transformer_encoder.layers.0.norm2.{bias, weight}
roi_heads.mask_head.mask_head_model.transformer_encoder.layers.0.self_attn.out_proj.{bias, weight}
roi_heads.mask_head.mask_head_model.transformer_encoder.layers.0.self_attn.{in_proj_bias, in_proj_weight}
WARNING [04/06 19:40:57 fvcore.common.checkpoint]: The checkpoint state_dict contains keys that are not used by the model:
  fc1000.{bias, weight}
  stem.conv1.bias


[04/06 19:40:57 d2.engine.train_loop]: Starting training from iteration 0
[04/06 19:41:04 d2.utils.events]:  eta: 1:14:04  iter: 19  total_loss: 5.908  loss_cls: 3.47  loss_box_reg: 0.0004418  loss_i_mask: 0.6926  loss_mask_bo: 0.1732  loss_mask: 0.6921  loss_justify: 0.1729  loss_rpn_cls: 0.688  loss_rpn_loc: 0.03921  time: 0.2247  data_time: 0.0313  lr: 9.9905e-06  max_mem: 2571M
[04/06 19:41:09 d2.utils.events]:  eta: 1:14:47  iter: 39  total_loss: 3.425  loss_cls: 1.122  loss_box_reg: 0.0005505  loss_i_mask: 0.6788  loss_mask_bo: 0.1811  loss_mask: 0.6794  loss_justify: 0.1184  loss_rpn_cls: 0.6736  loss_rpn_loc: 0.04445  time: 0.2260  data_time: 0.0092  lr: 1.998e-05  max_mem: 2571M
[04/06 19:41:14 d2.utils.events]:  eta: 1:16:01  iter: 59  total_loss: 2.409  loss_cls: 0.1636  loss_box_reg: 0.0049  loss_i_mask: 0.6686  loss_mask_bo: 0.1813  loss_mask: 0.6674  loss_justify: 0.03793  loss_rpn_cls: 0.6416  loss_rpn_loc: 0.04501  time: 0.2338  data_time: 0.0090  lr: 2.997e-05  max_mem: 2713M
[04/06 19:41:19 d2.utils.events]:  eta: 1:16:43  iter: 79  total_loss: 2.391  loss_cls: 0.2215  loss_box_reg: 0.01481  loss_i_mask: 0.6765  loss_mask_bo: 0.1679  loss_mask: 0.6766  loss_justify: 0.01452  loss_rpn_cls: 0.5916  loss_rpn_loc: 0.03428  time: 0.2373  data_time: 0.0095  lr: 3.9961e-05  max_mem: 3098M

**Comment by trqminh** [comment]
Sorry for the late response.
Actually their update_train2020 json file seems not reliable since there is no instance of the class rider. Therefore, I trained on their previous annotation files are instances_train.json and instances_val.json.
![image](https://user-images.githubusercontent.com/30286786/233472572-c4e74845-e58f-46ba-ae61-78ef8d2728a7.png)

Hope this helps.

**Comment by ZitengXue** [comment]
THANKS A LOT!

**Comment by JUFEhongren** [comment]
您好，我也遇到这个问题了，请问您解决了吗？

**Comment by Glaciercoool** [comment]
> 您好，我也遇到这个问题了，请问您解决了吗？

我也遇到这个问题，请问您解决了吗？

**Comment by Glaciercoool** [comment]
朋友，请问这个问题你之前解决了吗？我也还一直都是0...
